src/compiler/checker.ts

```diff
@@ -22201,8 +22209,8 @@ namespace ts {
         function checkObjectLiteralDestructuringPropertyAssignment(objectLiteralType: Type, property: ObjectLiteralElementLike, allProperties?: NodeArray<ObjectLiteralElementLike>, rightIsThis = false) {
             if (property.kind === SyntaxKind.PropertyAssignment || property.kind === SyntaxKind.ShorthandPropertyAssignment) {
                 const name = property.name;
-                const text = getTextOfPropertyName(name);

```

Comment:
for example, it would be nice to not have to change this code, and get `undefined` for names for which isComputedNonLiteralName is true.